Sources/Build/BuildDelegate.swift

```diff
@@ -90,7 +90,7 @@ final class TestDiscoveryCommand: CustomLLBuildCommand {
         }
 
         let outputs = tool.outputs.compactMap{ try? AbsolutePath(validating: $0.name) }
-        let testsByModule = Dictionary(grouping: tests, by: { $0.module })
+        let testsByModule = Dictionary(grouping: tests, by: { $0.module.spm_mangledToC99ExtendedIdentifier() })
```

Comment:
this fixes a bug were test discovery is broken for module names with "special" characters

Reply:
Makes sense.  As a separate PR I think we should probably also add the module name as a property on the target and consolidate the creation of a C99 compatible name to one place, and ideally also allow it to be customized if the default transformation isn't what the author wants.  But that's of course for later, and I might do that in support of a different PR I'm working unless there's objection.